### ISSUE TITLE
EVENT_AFTER_SAVE_ADDRESS fires multiple times when saving an address from the front end.

### ISSUE THREAD

**Issue by thisisjamessmith** [craftcms/commerce#1402]
### Description

The EVENT_AFTER_SAVE_ADDRESS event fires 3 times for addresses that are marked as primary and twice for addresses that are not when saving addresses using a front end form with the `commerce/customer-addresses/save` controller action.

### Steps to reproduce

1. Have a logged-in customer with some defined Commerce addresses
1. Have a plugin/module listening for EVENT_AFTER_SAVE_ADDRESS
1. Edit those addresses using a form posting with `{{ actionInput('commerce/customer-addresses/save') }}`
1. See the event fire multiple times in a single submission.

### Additional info

If it helps at all, I think the duplicate calls to `saveAddress` originate here in the order's `afterSave` method: https://github.com/craftcms/commerce/blob/9d31bbf73ea1aacc1f4148f63a944a24bc575f81/src/elements/Order.php#L1684-L1715

- in this context though, there is not an 'order' that requires saving - we're just updating a user's address book, so I'm not sure why this method is even firing.

As a workaround I'm simply setting a static flag in my own code to ensure it only fires once:

```
Event::on(
    Addresses::class,
    Addresses::EVENT_AFTER_SAVE_ADDRESS,
    function (Event $e) {
        // set a static flag to avoid multiple calls
        static $done = false;
        if (!$e->isNew && !$done) {
            
            // notify CRM system of address changes here
            ...

            // and don't do this again...
            $done = true;
        }
    }
);
```

- Craft version: 3.4.13, Commerce: 3.1.0.1

**Comment by lukeholder** [comment]
We will be fixing this in 4.0 when we refactor the order saving when saving an address which is a little awkward at the moment.

**Comment by lukeholder** [comment]
Fixed as of 4.0
